core/src/main/scala/ManifestValidator.scala

```diff
@@ -208,33 +208,65 @@ object ManifestValidator {
     }
 
     def validateReferencesDefaultNamespace(m: Manifest, defaultNamespace: NamespaceName): IO[Valid[Unit]] =
-    IO.pure(
-      m.namespaces.find(x => x.name == cfg.defaultNamespace)
-        .fold[Valid[Unit]](MissingDefaultNamespaceReference(cfg.defaultNamespace).invalidNel) { n =>
-          if (m.units.forall {u => n.units.exists(_._1 == u.name)})
-            ().validNel
-          else
-            MissingIndividualDefaultNamespaceReference(cfg.defaultNamespace).invalidNel
-        }
-    )
-
+      IO.pure(
+        m.namespaces.find(x => x.name == cfg.defaultNamespace)
+          .fold[Valid[Unit]](MissingDefaultNamespaceReference(cfg.defaultNamespace).invalidNel) { n =>
+            if (m.units.forall {u => n.units.exists(_._1 == u.name)})
+              ().validNel
+            else
+              MissingIndividualDefaultNamespaceReference(cfg.defaultNamespace).invalidNel
+          }
+      )
 
     def validateLoadbalancers(m: Manifest, dcs: Seq[Datacenter]): IO[Valid[Unit]] = {
       IO {
-        m.loadbalancers.foldLeft(Nil : List[Valid[Unit]])((res,lb) =>
+        m.loadbalancers.foldLeft(Nil: List[Valid[Unit]])((res,lb) =>
           validateLoadbalancer(lb, m.units, cfg.proxyPortWhitelist)(cfg.storage) :: res)
           .sequence
           .map(l => Foldable[List].fold(l))
       }
     }
 
+    def validateBlueprints(m: Manifest): IO[(Valid[Unit], List[Plan])] = {
+      val foooo: IO[List[Valid[Plan]]] = m.plans.foldLeft(IO.pure(List.empty[Valid[Plan]])) { (res, plan) =>
+        val result: IO[Valid[Plan]] = plan.environment.blueprint match {
+          // supplied manifest held a manifest reference
+          case Some(Left((ref,revision))) => {
```

Comment:
For the sake of type safety do we want to try the type parameter trick here? Where we have like:

```scala
case class Plan[A](...)
```

and set the type parameter to like `Unit` if its unsaturated and `Blueprint` if it is saturated or something? That way we can ensure in the scheduler we never get an unsaturated `Blueprint`

Reply:
Ah yeah fair, let's hold off on it for now then. Can we then modify the bits of code in the scheduler impls that currently call `.right.toOption` to instead actually inspect `Left(_)` and `Right(_)` and error out if it's a `Left(_)` so we don't silently "fail" there?